cavity_display.py

```diff
@@ -0,0 +1,73 @@
+from functools import partial
+
+from PyQt5.QtGui import QColor, QCursor
+from PyQt5.QtWidgets import (
+    QHBoxLayout,
+    QVBoxLayout,
+    QFrame,
+    QPushButton,
+    QGroupBox,
+)
+from pydm import Display
+from pydm.utilities import IconFont
+from pydm.widgets import PyDMByteIndicator, PyDMLabel
+
+from frontend.decoder import DecoderDisplay
+from frontend.gui_machine import GUIMachine
+from frontend.utils import make_line
+from lcls_tools.common.frontend.display.util import showDisplay
+
+
+class CavityDisplayGUI(Display):
+    def __init__(self, parent=None, *args, **kwargs):
+        super().__init__(parent, *args, **kwargs)
+        self.setStyleSheet(
+            "background-color: rgb(35, 35, 35); color: rgb(255, 255, 255); font-size: 15pt;"
+        )
+
+        self.gui_machine = GUIMachine()
+
+        self.header = QHBoxLayout()
+        heartbeat_indicator = PyDMByteIndicator(
+            init_channel="ALRM:SYS0:SC_CAV_FAULT:ALHBERR"
+        )
+        heartbeat_indicator.onColor = QColor(255, 0, 0)
+        heartbeat_indicator.offColor = QColor(0, 255, 0)
+        heartbeat_indicator.showLabels = False
+        heartbeat_indicator.circles = True
+        heartbeat_indicator.showLabels = False
+
+        heartbeat_label = PyDMLabel(init_channel="ALRM:SYS0:SC_CAV_FAULT:ALHBERR")
+        heartbeat_counter = PyDMLabel(init_channel="PHYS:SYS0:1:SC_CAV_FAULT_HEARTBEAT")
+
+        self.header.addWidget(heartbeat_indicator)
+        self.header.addWidget(heartbeat_label)
+        self.header.addWidget(heartbeat_counter)
+        self.header.addStretch()
+
+        self.decoder_window: DecoderDisplay = DecoderDisplay()
+
+        self.decoder = QPushButton("Three Letter Code Decoder")
+        self.decoder.clicked.connect(partial(showDisplay, self.decoder_window))
+
+        icon = IconFont().icon("file")
+        self.decoder.setIcon(icon)
+        self.decoder.setCursor(QCursor(icon.pixmap(16, 16)))
+        self.decoder.openInNewWindow = True
+        self.header.addWidget(self.decoder)
+
+        self.setWindowTitle("SRF Cavity Display")
+
+        self.vlayout = QVBoxLayout()
+        self.vlayout.setContentsMargins(0, 0, 0, 0)
+        self.groupbox_vlayout = QVBoxLayout()
+        self.groupbox_vlayout.addLayout(self.header)
+        self.setLayout(self.vlayout)
+
+        self.groupbox_vlayout.addLayout(self.gui_machine.top_half)
+        self.groupbox_vlayout.addWidget(make_line(QFrame.HLine))
```

Comment:
Between line 67 and 68, would we like to add 
`self.groupbox_vlayout.addSpacing(10)`
so that there is a space between the cavity 8 status bars and the horizontal line  above L3B? Aka more visual separation?